apps/04TowersOfHanoi.js

```diff
@@ -18,51 +18,33 @@ function printStacks() {
     console.log("c: " + stacks.c);
 }
 
-function movePiece(startStack, endStack) {
-    var move = stacks[startStack].pop();
-    stacks[endStack].push(move);
-};
-//
-function isLegal(startStack, endStack){
-  if (stacks[endStack].length === 0){
+function movePiece (startStack, endStack) {
+  if (isLegal(startStack, endStack)=== true) {
+    stacks[endStack].push(stacks[startStack].pop());
+  }
+}
+
+function isLegal (startStack, endStack) {
+  if (stacks[endStack].length === 0) {
     return true;
```

Comment:
What if I made a mistake and in the beginning of the game i chose b and c, stacks that are both empty?